beer-garden/brewtils
# #334 - Updating sphinx generated docs

Closes #334 

This attempts to get the sphinx generated docs up to date with the current state of brewtils.  Things addressed in this PR:

* Resolved all sphinx build warnings
* Generated new .rst files so that all current modules are represented (e.g. pika).
* Updated some docstrings that did not conform to the Google style.
* Fixed miscellaneous formatting errors.
* Added deprecation warnings where appropriate.
* Minor updates to the contribution guide.

Things specifically **not** done:
* Make sure type information on all the arguments and returns are correct. That would be quite the exercise at this point, so it's probably better to chip away at it over time.

I don't expect that this PR makes the documentation is 100% accurate.  The goal is just to make it more closely reflect reality and clean up obvious issues where I could.

# Test Instructions
* Check out branch `issue/334`, make sure you have a properly configured virtualenv, etc.
* Make sure you can build the docs without errors or warnings:
  ```shell
  make clean-docs && make docs
  ```
* You should then be able to view the files generated docs in a browser by navigating to the build folder.  Something like: `file:///path/to/brewtils/docs/_build/html/index.html`
* From there, click around and make sure things look good and work right. If you see anything in the PR diff that is unclear how it would render, be sure to find that spot in the docs and see how it turned out.
